tests/e2e/export.go

```diff
@@ -15,17 +15,57 @@
 package e2e
 
 import (
+	"context"
+	"fmt"
 	"path/filepath"
 	"strings"
 
 	"github.com/GoogleCloudPlatform/k8s-config-connector/config/tests/samples/create"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/cli/cmd/export"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/yaml"
 )
 
+func directExport(u *unstructured.Unstructured, c client.Client) ([]byte, error) {
+	ctx := context.TODO()
+	gvk := u.GroupVersionKind()
+	model, err := registry.GetModel(gvk.GroupKind())
+	if err != nil {
+		return nil, err
+	}
+	a, err := model.AdapterForObject(ctx, c, u)
+	if err != nil {
+		return nil, err
+	}
+
+	found, err := a.Find(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, fmt.Errorf("GVK %s '%v' is not found", u.GroupVersionKind(), u.GetName())
```

Comment:
Nit: a message like `"GCP resource for %v with name %q not found", gvk.Kind, u.GetName` might be clearer.  My thought is that it's not the GVK that wasn't found (in the registry), it's the underlying GCP object that wasn't found.